### PR TITLE
Make sure header name `Content-MD5` is lower-cased

### PR DESCRIPTION
**Pull request by evanxg852000**
Should close [this](https://github.com/Azure/azure-sdk-for-rust/issues/928)

### CODE REVIEW

File: sdk/storage_blobs/src/blob/operations/put_block_list.rs [hunk 1]
```diff
@@ -37,7 +37,7 @@ impl PutBlockListBuilder {
             };
 
             let mut headers = Headers::new();
-            headers.insert("Content-MD5", &md5);
+            headers.insert("content-md5", &md5);
```
Comment (bmc-msft) [inline sdk/storage_blobs/src/blob/operations/put_block_list.rs:40]:
```suggestion
            headers.insert(CONTENT_MD5, &md5);
```

**Review by bmc-msft — CHANGES_REQUESTED** [review]
Please use the HeaderName value provided by `azure_core::headers::CONTENT_MD5`

Additionally, instead of a hard-coded blob name, can this be a parameter to the example?

File: sdk/storage_blobs/examples/put_block_blob_00.rs [hunk 2]
```diff
@@ -23,9 +23,31 @@ async fn main() -> azure_core::Result<()> {
...
 
-    let blob_client = StorageClient::new_access_key(&account, &access_key)
-        .container_client(&container)
-        .blob_client(&blob_name);
+    let container_client =
+        StorageClient::new_access_key(&account, &access_key).container_client(&container);
+
+    //put list of block
+    {
+        let foo_blob_client = container_client.blob_client("foo");
```
[the remark this answers is not in the document]
Comment (evanxg852000) [reply sdk/storage_blobs/examples/put_block_blob_00.rs]: I added a separate example file to demonstrate multi-part blob upload which was not possible.

**Comment by bmc-msft** [comment]
Thanks for the contribution!

**Pull request merged**